assets/src/console/gql/schema.graphql

```diff
@@ -5,6 +5,7 @@ type Query {
   gatewayAccount(id: ID!): GatewayAccount!
   products(gatewayAccountId: ID!): [Product!]!
   payments(gatewayAccountId: ID!): [Payment!]!
+  paymentEvents(paymentId: ID!): [PaymentEvent!]!
```

Comment:
i would have expected this to nest within payment

```
payment(id) {
  amount
  ...
  events {
    status
  }
}
```

Reply:
@dagoof yeah me too, but we can't because the `/pay/:id`routes are handled in parent components and that's where it fetches the payment object. so in this case it made sense to just fetch the events in the page that needs them. it's probably a problem with the design of having a parent component making the query and assuming that no child component will want other data in the query. definitely worth looking at to see if there's a better way to compose things so you get the benefit of shared parent logic but children can provide a different query, for example.